### PR TITLE
Update image capture sample

### PR DESCRIPTION
**Pull request by beaufortfrancois**
R: @miguelao 
Preview at https://beaufortfrancois.github.io/samples/image-capture/

### CODE REVIEW

File: image-capture/update-camera-zoom.js [hunk 1]
```diff
@@ -3,24 +3,24 @@ navigator.mediaDevices.getUserMedia({video: true})
...
-    input.min = photoCapabilities.zoom.min;
-    input.max = photoCapabilities.zoom.max;
-    input.step = photoCapabilities.zoom.step;
-    input.value = photoCapabilities.zoom.current;
-    input.oninput = function(event) {
-      imageCapture.setOptions({zoom: event.target.value});
-    }
-  });
+  // Check whether zoom is supported or not.
+  if (!capabilities.zoom.min && !capabilities.zoom.max) {
```
Comment (yell0wd0g) [inline image-capture/update-camera-zoom.js]: This should be `if (capabilities.zoom)`

File: image-capture/update-camera-zoom.js [hunk 2]
```diff
@@ -3,24 +3,24 @@ navigator.mediaDevices.getUserMedia({video: true})
...
+    return Promise.reject('Zoom is not supported by ' + track.label);
+  }
+
+  // Map zoom to a slider element.
+  input.min = capabilities.zoom.min;
+  input.max = capabilities.zoom.max;
+  input.step = capabilities.zoom.step;
+  input.value = settings.zoom;
+  input.oninput = function(event) {
+    track.applyConstraints({zoom: event.target.value});
```
Comment (yell0wd0g) [inline image-capture/update-camera-zoom.js]: `track.applyConstraints({ advanced : [{zoom: event.target.value}] });`

File: image-capture/update-camera-zoom.js (hunk 1 above)
Comment (beaufortfrancois) [reply image-capture/update-camera-zoom.js]: @miguelao Please update example from https://www.w3.org/TR/image-capture/#update-camera-zoom-and-code-takephoto-code- then.
Thanks!

File: image-capture/update-camera-zoom.js (hunk 2 above)
Comment (beaufortfrancois) [reply image-capture/update-camera-zoom.js]: @miguelao Please update example from https://www.w3.org/TR/image-capture/#update-camera-zoom-and-code-takephoto-code- then.
Thanks!

File: image-capture/update-camera-zoom.js [hunk 3]
```diff
@@ -1,26 +1,35 @@
 navigator.mediaDevices.getUserMedia({video: true})
-.then(mediaStream => {
+.then(async mediaStream => {
   document.querySelector('video').srcObject = mediaStream;
 
+  // Needed because getCapabilities may be null if accessed too early...
```
Comment (yell0wd0g) [inline image-capture/update-camera-zoom.js]: Please mention here the bug [crbug.com/711524](https://crbug.com/711524) and perhaps explain that capabilities/settings can only be retrieved after the device starts streaming, and that this happens after and asynchronously w.r.t. `getUserMedia()` return.

File: image-capture/grab-frame-take-photo.js [hunk 4]
```diff
@@ -32,11 +32,15 @@ function onTakePhotoButtonClick() {
 
 /* Utils */
 
-function drawCanvas(canvas, imageBitmap) {
-  canvas.width = imageBitmap.width;
-  canvas.height = imageBitmap.height;
-  canvas.getContext('2d').drawImage(imageBitmap, 0, 0,
-      imageBitmap.width, imageBitmap.height);
+function drawCanvas(canvas, img) {
```
Comment (yell0wd0g) [inline image-capture/grab-frame-take-photo.js:35]: This function doesn't seem to be used.

**Review by yell0wd0g — APPROVED** [review]
lgtm % 2 comments!

File: image-capture/grab-frame-take-photo.js (hunk 4 above)
Comment (beaufortfrancois) [reply image-capture/grab-frame-take-photo.js:35]: It is called every time you press "Take Photo" buttron and "Grab Frame" button. See https://beaufortfrancois.github.io/samples/image-capture/grab-frame-take-photo.html

File: image-capture/update-camera-zoom.js (hunk 3 above)
Comment (beaufortfrancois) [reply image-capture/update-camera-zoom.js]: Done.

**Comment by beaufortfrancois** [comment]
I've merged this as https://developers.google.com/web/updates/2016/12/imagecapture is now live.

**Pull request merged**